src/sql/parts/modelComponents/tree/treeComponentRenderer.ts

```diff
@@ -141,8 +141,9 @@ export class TreeComponentRenderer extends Disposable implements IRenderer {
 		dom.toggleClass(templateData.icon, 'model-view-tree-node-item-icon', !!icon);
 		if (element) {
 			element.onCheckedChanged = (checked: boolean) => {
+				element.checked = checked;
```

Comment:
I don't understand if the element is already had changed the checked value so the onCheckChanged is raised, why we have to set that again? can you explain the change please?

Reply:
Even though the checkbox of the element in UI is checked by user, the `checked` field of the element object is not changed. And it requires this explicit assignment to force to update the `element` object. If you think the `checked` status of `element` object should be automatically updated when user clicks the checkbox in UI, and this assignment (`element.checked = checked;`) should not be needed, can you please fix it so that this PR can be skipped?